src/flownet/network_model/_generate_connections.py

```diff
@@ -94,9 +94,51 @@ def _remove_recorded_connections(
     return conn_matrix
 
 
+def _split_additional_flow_nodes(
+    total_additional_nodes: int, concave_hull_list: List[np.ndarray]
+) -> List:
+    """
+    This function takes splits the additional_flow_nodes defined in the config over the layers.
+    The division is based on the size (volume) of the boundingbox of the layer.
+
+    Args:
+        configuration: FlowNet configuration yaml as dictionary
+        concave_hull_list: List of boundingbox per layer, i.e., numpy array with x, y, z min/max
+            boundingboxes for each grid block
+
+    Returns:
+        List of additional notes per layer (len of list same as number of layers)
+
+    """
+    volumes = []
+    for xyz in concave_hull_list:
+        if len(xyz.shape) == 1:
+            num_bounding_boxes = 1
+            xyz = xyz[None, :]
+        else:
+            num_bounding_boxes = xyz.shape[0]
+        volume = sum(
+            [
+                (
+                    (xyz[i, 1] - xyz[i, 0])
+                    * (xyz[i, 3] - xyz[i, 2])
+                    * (xyz[i, 5] - xyz[i, 4])
+                )
+                for i in range(0, num_bounding_boxes)
+            ]
+        )
+        volumes.append(volume)
+
+    fractions = [v / sum(volumes) for v in volumes]
+    sep_add_flownodes = [round(frac * total_additional_nodes) for frac in fractions]
+
+    return sep_add_flownodes
+
+
 def _generate_connections(
     df_coordinates: pd.DataFrame,
     configuration: Any,
+    additional_flow_nodes: int,
```

Comment:
Why the change in name: additional **_flow** _nodes?

Reply:
Because you suggested that earlier on ;) 
https://github.com/equinor/flownet/pull/305#discussion_r561673669